### PR TITLE
Refactor DeleteButton to not display when snippet has not been saved

### PR DESCRIPTION
**Pull request by solkaz**
## Description
What the title says

## Motivation and Context
Per @thebho's request

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Browsers

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Mobile:
- [ ] Other:

## Screenshots (if appropriate):
screen shot 2017-05-25 at 10 38 37 am

screen shot 2017-05-25 at 10 37 46 am

screen shot 2017-05-25 at 10 38 01 am

### CODE REVIEW

File: src/components/buttons/DeleteButton.jsx [hunk 1]
```diff
@@ -16,20 +16,10 @@ const styles = {
...
-      Click to delete snippet
-    </span>)
-    :
-    (<span>
-      You can only delete snippets
-      <br />
-      that you own
-    </span>)
-);
+const hideIfNotSaved = isNotSaved => branch(isNotSaved, renderNothing);
```
Comment (dane-johnson) [inline src/components/buttons/DeleteButton.jsx]: You should refactor to use positive booleans

File: src/components/buttons/DeleteButton.jsx [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import React, { PropTypes } from 'react';
-
 import IconButton from 'material-ui/IconButton';
 import TrashCan from 'material-ui/svg-icons/action/delete';
 import ReactTooltip from 'react-tooltip';
+import { branch, renderNothing } from 'recompose';
```
Comment (dane-johnson) [inline src/components/buttons/DeleteButton.jsx]: We already have a pattern for this functionality
```
{(Saved) ? <Component/> : null}
```

Comment (solkaz) [inline src/components/buttons/DeleteButton.jsx]: I know that's an established pattern, but it's not one that has benefit over this method. Recompose helps make reasoning about components easier, it's an essential for FP in React, and it's more readable vs. the ternary operator.

Comment (dane-johnson) [inline src/components/buttons/DeleteButton.jsx]: The benefit is that it is established. At this point we all know what it means, and it's littered throughout the project. We shouldn't have two patterns to do the same thing.

Comment (solkaz) [inline src/components/buttons/DeleteButton.jsx]: We're done working on this project next week, and I don't expect anyone to have to edit this component soon.

**Pull request closed without merging**